src/firestarter/Environment/X86/Payload/AVX512Payload.cpp

```diff
@@ -158,6 +179,42 @@ auto AVX512Payload::compilePayload(const environment::payload::PayloadSettings&
   for (auto const& Reg : ShiftReg32) {
     Cb.mov(Reg, Imm(0xAAAAAAAA));
   }
+
+  // Init AMX registers and config
+  __tilecfg tile_data = {0};
+  request_permission();
+  create_AMX_config(&tile_data); // Create tilecfg and fill it
+
+  static bool init = true;
+  uintptr_t src1, src2;
+  uint64_t src3;
+  unsigned int aligned_alloc_size = static_cast<unsigned int>(MAX * sizeof(__bfloat16));
+  if (aligned_alloc_size % 1024) { // aligned_alloc expects size to be multiple of alignment (aka 1024)
+    aligned_alloc_size = aligned_alloc_size + (1024 - (aligned_alloc_size % 1024));
+  }
+  src1 = (uintptr_t)aligned_alloc(1024, aligned_alloc_size);
+  src2 = (uintptr_t)aligned_alloc(1024, aligned_alloc_size);
+  src3 = (uint64_t)aligned_alloc(1024, aligned_alloc_size);
+  if (((void*)src1 == nullptr) || (void*)src2 == nullptr ||
+      (void*)src3 == nullptr) { // uintptr_t garantuees we can cast it to void* and back
+    std::cout << "[ERROR]: Allocation of source and target buffer for AMX failed. Aborting...\n";
+    exit(1);
+  }
```

Comment:
This memory should be allocated in the `LoadWorkerMemory` class. A platform independent alligned alloc abstraction is available there. You might need to change it sightly, so that these variables are aligned with 1024B instead of 64B. This change will not only allocate these arrays for the AVX512/AMX payload but for all. There should however be no negative effect other than increased allocated RAM size for all payloads.